remove unneeded lines from pass otp output for HOTP

fstecker opened the pull request:
When using the --otp feature of qute_pass, it copies the entire output of "pass otp website.com/account" into the text field. For HOTP this includes two unwanted lines, for example:

[master 1af8998] Increment HOTP counter for website.com/account.
 1 file changed, 0 insertions(+), 0 deletions(-)
123456

As a simple fix, I changed the pass_otp function to only return the last line of the output. This should not make a difference for other working cases like TOTP, where the output of pass otp is a simple number of up to 10 digits.

The-Compiler commented:
cc @murchik @bascht @lufte

lufte commented:
I can't provide feedback on this one as I don't run pass-otp. Although it does sound weird that you get what looks like git's output mixed with the password.

fstecker commented:
> I can't provide feedback on this one as I don't run pass-otp. Although it does sound weird that you get what looks like git's output mixed with the password.

pass otp first updates the URL in git, incrementing the counter by one, then outputs the code. So this output is what you would expect. For the case that you just want the password, pass otp also offers the --clip option, which copies the code to the clipboard instead of printing it to stdout. qute_pass could use that, but I don't like it: it would overwrite the clipboard, and potentially leak the code to other applications. This fix is simpler and avoids that.

lufte commented:
> pass otp first updates the URL in git, incrementing the counter by one, then outputs the code. So this output is what you would expect. For the case that you just want the password, pass otp also offers the --clip option, which copies the code to the clipboard instead of printing it to stdout. qute_pass could use that, but I don't like it: it would overwrite the clipboard, and potentially leak the code to other applications. This fix is simpler and avoids that.

I see, that makes sense. And I agree that this is better than using the clipboard.

bascht commented:
@fstecker I use `gopass` as well, so I cannot really reproduce - but I had a quick look into the `pass-otp` implementation and it looks like the garbled output can be suppressed:

```bash
  if [[ "$quiet" -eq 1 ]]; then
    git_add_file "$passfile" "$message" 1>/dev/null
  else
    git_add_file "$passfile" "$message"
  fi
```

https://github.com/tadfisher/pass-otp/blob/develop/otp.bash#L133-L149

It looks like the wrapper [will accept `--quiet`](https://github.com/tadfisher/pass-otp/blob/develop/otp.bash#L325) - have you tried that, instead of `--clip`?

fstecker commented:
> It looks like the wrapper [will accept `--quiet`](https://github.com/tadfisher/pass-otp/blob/develop/otp.bash#L325) - have you tried that, instead of `--clip`?

This was a somewhat recent addition to pass-otp (from 2020, but the latest release 1.2.0 is from 2018):
https://github.com/tadfisher/pass-otp/commit/99419824e12ccfa082967248df356e15f456e8cd
https://github.com/tadfisher/pass-otp/blob/develop/CHANGELOG.md

I have version 1.2.0 installed, and get this:
```
$ pass otp --quiet website.com/username
pass: unrecognized option '--quiet'
Usage: pass otp [--clip,-c] pass-name
```

So here are the options I can see:
1. Use my workaround (but perhaps add a comment that this might be unnecessary in the future)
2. Supply the `--quiet` option and break it for everyone not using the current git version of pass-otp
3. Somehow query the pass-otp version and use --quiet if available
4. Ask the pass-otp creator(s) to release a new version?

I'm happy to do any of them, you can choose.

bascht commented:
@fstecker Oh the joys of upstream packaging :D 

I think (1) is the way to go then. A comment referring to this issue or the not yet released switch in `pass-otp` would be good. 

Maybe you can also ping the maintainer to tag another release, there is quite the feature-buildup:
https://github.com/tadfisher/pass-otp/compare/v1.2.0...develop

fstecker commented:
@basht just to be sure, gopass does not show this behaviour, even for HOTP?

I now changed it to only apply if pass is being used, and added a comment.
